### PR TITLE
feat(docfx): add support for docfx

### PR DESCRIPTION
**Pull request by dmccaffery**
* add `GetLatestRelease` task used to retrieve a release asset from a GitHub repository
* add automatic download of DocFX if not already available on the system
* add detection of a command / executable path
* detect mono on *nix systems, which is a prerequisite for docfx
* add support for detecting and building docfx projects

Closes: #21

### CODE REVIEW

**Comment by dmccaffery** [comment]
@automotiveMastermind/masterminds this is the beginnings of adding documentation for condo, as well as supporting automatic documentation of aM projects.

Mono is a prerequisite of `DocFX` for the time being, and must be installed outside of condo if creating a build on *nix platforms. Mono is a fairly invasive framework and is installed in many different ways on different flavors of Linux, so it just doesn't make sense to install it within the condo-lifecycle just to support doc generation; especially since Travis, CircleCI, and VSTS all support mono on their agents.

**Comment by falquan** [comment]
I continue with my belief that Mono is just a disease! :)

File: src/AM.Condo/Tasks/GetLatestRelease.cs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        /// </summary>
+        public string Token { get; set; }
+        #endregion
+
+        #region Methods
+        /// <inheritdoc />
+        public override bool Execute()
+        {
+            // attempt to create the root API URL
+            if (!Uri.TryCreate(this.GitHubApi, UriKind.Absolute, out Uri baseUri))
```
Comment (falquan) [inline src/AM.Condo/Tasks/GetLatestRelease.cs:67]: Has nothing to do with this, but curious on ideas on how to more elegantly handle TryParse/TryXYZ situations.

**Review by falquan — APPROVED** [review]

**Comment by dmccaffery** [comment]
I cannot merge, as the 2.16 release of docfx (posted 6 hours ago) broke something:

dotnet/docfx#1559

**Comment by dmccaffery** [comment]
I am going to require docs to be generated on Windows so I can get this through -- want to use it for aM work.

**Pull request merged**